test/hooks_test.dart

```diff
@@ -92,5 +92,171 @@ main() {
         expect(countRef.text, '1');
       });
     });
+
+    group('useEffect -', () {
+      ReactDartFunctionComponentFactoryProxy UseEffectTest;
+      ButtonElement countButtonRef;
+      DivElement countRef;
+      DivElement mountNode;
+      int useEffectCallCount;
+      int useEffectCleanupCallCount;
+      int useEffectWithDepsCallCount;
+      int useEffectCleanupWithDepsCallCount;
+      int useEffectWithDepsCallCount2;
+      int useEffectCleanupWithDepsCallCount2;
+      int useEffectWithEmptyDepsCallCount;
+      int useEffectCleanupWithEmptyDepsCallCount;
+
+      setUpAll(() async {
+        mountNode = new DivElement();
+        useEffectCallCount = 0;
+        useEffectCleanupCallCount = 0;
+        useEffectWithDepsCallCount = 0;
+        useEffectCleanupWithDepsCallCount = 0;
+        useEffectWithDepsCallCount2 = 0;
+        useEffectCleanupWithDepsCallCount2 = 0;
+        useEffectWithEmptyDepsCallCount = 0;
+        useEffectCleanupWithEmptyDepsCallCount = 0;
+
+        UseEffectTest = react.registerFunctionComponent((Map props) {
+          final count = useState(0);
+          final countDown = useState(0);
+
+          useEffect(() {
+            useEffectCallCount++;
+            return () {
+              useEffectCleanupCallCount++;
+            };
+          });
+
+          useEffect(() {
+            useEffectWithDepsCallCount++;
+            return () {
+              useEffectCleanupWithDepsCallCount++;
+            };
+          }, [count.value]);
+
+          useEffect(() {
+            useEffectWithDepsCallCount2++;
+            return () {
+              useEffectCleanupWithDepsCallCount2++;
+            };
+          }, [countDown.value]);
+
+          useEffect(() {
+            useEffectWithEmptyDepsCallCount++;
+            return () {
+              useEffectCleanupWithEmptyDepsCallCount++;
+            };
+          }, []);
+
+          return react.div({}, [
+            react.div({
+              'ref': (ref) {
+                countRef = ref;
+              },
+            }, [
+              count.value
+            ]),
+            react.button({
+              'onClick': (_) {
+                count.set(count.value + 1);
+              },
+              'ref': (ref) {
+                countButtonRef = ref;
+              },
+            }, [
+              '+'
+            ]),
+          ]);
+        });
+
+        react_dom.render(UseEffectTest({}), mountNode);
+
+        await pumpEventQueue();
```

Comment:
This should have a comment saying why it's necessary

Reply:
Apparently I didn't need them anymore haha